src/lib/retire-js/RetireJsEngine.ts

```diff
@@ -160,23 +166,44 @@ export class RetireJsEngine implements RuleEngine {
 					// Convert the output into RuleResult objects and resolve to that.
 					res(this.processOutput(stderr, invocation.rule));
 				} else {
-					// If RetireJS exits with any other code, then it means something went wrong. We'll reject with stderr
-					// for the ease of upstream error handling.
-					rej(stderr);
+					// If RetireJS exits with any other code, then it means something went wrong. The error could be
+					// contained in either stdout or stderr, so we'll send them both to a method for processing, and
+					// reject the string we receive after prefixing it with an informative header.
+					this.logger.trace(`RetireJS failed with exit code ${code}`);
```

Comment:
Thanks for adding the log. My recommendation for log level would be `warning` since these deal with error conditions.